beamform/craftcor_tab.py

```diff
@@ -126,30 +237,175 @@ def do_f_tab(self, corr, iant):
+        nsamp = corr.nint * corr.nfft               # number of samples in 1.185 MHz oversampled channel
+        nchan_coarse = len(corr.freqs)              # number of coarse channels 
+        
+        sampoff = whole_delay + corr.abs_delay      # starting sample used in cropping antenna buffers, this is the earliest
+                                                    # sample for which there is data for all the antennas
 
-        # time-dependent geometric delays
-        # np.linspace(0, 1, nsamp) == time in units of integrations
+        # time-dependent geometric delays, these will be used to beamform the antennas
         geom_delays_us = (
             geom_delay_us
             + geom_delay_rate_us * np.linspace(0, 1, nsamp)
             - fixed_delay_us
         )
-        np.save("TEMP_geom_delays_us.npy", geom_delays_us)
-        del geom_delays_us
-        geom_delays_us = np.load("TEMP_geom_delays_us.npy", mmap_mode="r")
 
-        sampoff = whole_delay + corr.abs_delay
+
+        # do a bunch of printing
+        print(f"corr.refant.trigger_frame: {corr.refant.trigger_frame}")
+        print(f"self.trigger_frame: {self.trigger_frame}")
+        print(f"total_delay_samp: {total_delay_samp}")
+        print(f"whole_delay: {whole_delay}")
+        print(f"corr.abs_delay: {corr.abs_delay}")
+        
+
+
+        ###########################################################################
+        # Crop antenna buffers, use the DM and rough candidate position of the FRB
+        # to create a crop that encompasses the entire DM sweep of the FRB.
+        ###########################################################################
+
+        # before doing any cropping, enforce sampoff and nsamp to be multiples of 32, in case we request
+        # a crop outside the antenna buffer bounds, we can simply snap to these bounds safely knowing we aren't
+        # introducing any fractional delay.
+        # NOTE: under the assumption that the non-multiple of 32 error comes from a precision error, we are going to assume 
+        # this error is only 1 or 2 samples, hence we will round to the nearest multiple of 32, then add 32
+        old_sampoff = sampoff
+        if (sampoff % 32) != 0:
+            sampoff = round(sampoff / 32) * 32 + 32 # extra 32 to ensure we are within all antenna
+        sampoff_skip = sampoff - old_sampoff    # in the case that we have to slightly change the initial sampoff bound, this 
+        # difference can be used to skip ahead in the geometric delays.
+        
+        nsamp = (nsamp // 32) * 32 #
+        if sampoff + nsamp > self.vfile.nsamps:
+            # make sure these bounds are defined
+            nsamp -= 32
+
+        ### Now do cropping ###
+        if (mjd is not None) and (DM is not None):
+            print("FRB cropping\n")
+            
+            # sample number of candidate position
+            cand_offset_samp = int((mjd - corr.refant.mjdstart) * 8.64e10 * 32/27) + sampoff
```

Comment:
should this be rounded before being cast to an int? Is this a possible place that an off-by-one error could arise?

Reply:
yes